capstone/backend/src/test/java/com/google/CreateOrUpdateUserlistServletTest.java

```diff
@@ -0,0 +1,51 @@
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.BufferedReader;
+import java.io.StringWriter;
+import java.io.StringReader;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import junit.framework.TestCase;
+import org.mockito.Mockito;
+
+public class CreateOrUpdateUserlistServletTest extends TestCase {
+  public void testDoPostSucceed() throws ServletException, IOException {
+    HttpServletRequest req = Mockito.mock(HttpServletRequest.class);
+    HttpServletResponse res = Mockito.mock(HttpServletResponse.class);
+    StringWriter writer = new StringWriter();
+    StringReader reader = new StringReader(
+        "{\"userId\": 1,\"userList\":{\"listId\":2,\"displayName\":\"List Name\",\"itemTypes\":[\"Butter\",\"Orange Juice\", \"Peanuts\"]}");
+    CreateOrUpdateUserlistServlet servlet = new CreateOrUpdateUserlistServlet();
```

Comment:
Move this line to just before 'servlet.doPost'.

Also usually we divide a test function into three parts, separated by an empty line:
1. The mock data setup.
2. The actual call.
3. The verification steps.

If you follow this pattern, you will see that in part 1, the only difference between this and the test below is the StringReader parameter, so you can extract a common function 'setupMockData' and pass the string as a parameter.

Similarly part 2 can also be extracted into a common function.